Content.Server/_NF/Storage/AnchorableStorageSystem.cs

```diff
@@ -0,0 +1,122 @@
+        SubscribeLocalEvent<AnchorableStorageComponent, ContainerIsInsertingAttemptEvent>(OnInsertAttempt);
+        _anchorableStorageQuery = GetEntityQuery<AnchorableStorageComponent>();
+    }
+
+    private void OnAnchorStateChanged(Entity<AnchorableStorageComponent> ent, ref AnchorStateChangedEvent args)
+    {
+        if (!args.Anchored)
+            return;
+
+        if (CheckOverlap((ent, ent.Comp, Transform(ent))))
+        {
+            _popup.PopupEntity(Loc.GetString("anchored-storage-already-present"), ent);
+            _xform.Unanchor(ent, Transform(ent));
+            return;
+        }
+
+        // Eject any sapient creatures inside the storage.
+        // Hack: does not recurse down into bags in bags.
+        if (!TryComp(ent.Owner, out StorageComponent? storage))
+            return;
+        List<EntityUid> entsToRemove = new();
+        foreach (var storedItem in storage.StoredItems.Keys)
+        {
+            if (HasComp<MindContainerComponent>(storedItem) || HasComp<PseudoItemComponent>(storedItem))
+                entsToRemove.Add(storedItem);
+        }
+        foreach (var removeUid in entsToRemove)
+        {
+            _container.RemoveEntity(ent.Owner, removeUid);
+        }
+    }
+
+    private void OnAnchorAttempt(Entity<AnchorableStorageComponent> ent, ref AnchorAttemptEvent args)
+    {
+        if (args.Cancelled)
+            return;
+
+        if (CheckOverlap((ent, ent.Comp, Transform(ent))))
+        {
+            _popup.PopupEntity(Loc.GetString("anchored-storage-already-present"), ent, args.User);
+            args.Cancel();
+        }
+    }
+
+    private void OnInsertAttempt(Entity<AnchorableStorageComponent> ent, ref ContainerIsInsertingAttemptEvent args)
+    {
+        if (args.Cancelled)
+            return;
+
+        // Check for living things, they should not insert when anchored.
+        if (HasComp<MindContainerComponent>(args.EntityUid) || HasComp<PseudoItemComponent>(args.EntityUid))
+        {
+            if (Transform(ent.Owner).Anchored)
+                args.Cancel();
+        }
+    }
+
+    [PublicAPI]
+    public bool CheckOverlap(EntityUid uid)
+    {
+        if (!_anchorableStorageQuery.TryComp(uid, out var node))
+            return false;
+
+        return CheckOverlap((uid, node, Transform(uid)));
+    }
+
+    public bool CheckOverlap(Entity<AnchorableStorageComponent, TransformComponent> ent)
+    {
+        if (ent.Comp2.GridUid is not { } grid || !TryComp<MapGridComponent>(grid, out var gridComp))
+            return false;
+
+        var indices = _map.TileIndicesFor(grid, gridComp, ent.Comp2.Coordinates);
+        _anchoredEntities.Clear();
+        _map.GetAnchoredEntities((grid, gridComp), indices, _anchoredEntities);
+
+        foreach (var otherEnt in _anchoredEntities)
+        {
+            // Don't match yourself.
+            if (otherEnt == ent.Owner)
+                continue;
```

Comment:
You can use LINQ here.
`anchoredEntities.Where(otherEnt => otherEnt != ent.Owner)`

Reply:
I'll disagree with this one.  An enumerator would be better, we don't need to process an enumerable sequence into a list into another list.  Revision inbound.

Reply:
Fun fact: that enumerable sequence is also a list.

Reply:
Isn't the whole point of LINQ that it's as lazy as possible? You can chain .Where()s and .Select()s without creating intermediate collections, because it's all lazy IEnumerables all the way down. That said, there's a cost to allocating closures that's often overlooked, and sometimes a plain old `continue` is preferable.